src/layout/InsertionForm.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import { Form, Col, Row, Button, InputGroup } from 'react-bootstrap';
+
+function InsertionForm(props) {
+  const [amount, setAmount] = useState('');
+  const [category, setCategory] = useState('');
+  const [description, setDescription] = useState('');
+
+  function setValue(e) {
+    switch (e.target.name) {
+      case 'amount':
+        setAmount(e.target.value);
+        break;
+
+      case 'category':
+        setCategory(e.target.value);
+        break;
+
+      case 'description':
+        setDescription(e.target.value);
+        break;
+      default:
+        break;
+    }
+  }
+
+  function addIncomeButton() {
+    if (
+      amount !== '' ||
+      category !== '' ||
+      category !== 'Select a category' ||
+      description !== ''
+    ) {
+      props.addIncome(amount, category, description);
+      setAmount('');
+      setCategory('Travel');
+      setDescription('');
+    }
+  }
+
+  function addExpenseButton() {
+    if (amount !== '' || category !== '' || description !== '') {
+      props.addExpense(amount, category, description);
+      setAmount('');
+      setCategory('Travel');
+      setDescription('');
+    }
+  }
+
+  return (
+    <Col sm={12} md={6} lg={4} className="mt-4 ms-3">
+      <Form>
+        <h4>Add a new record</h4>
+        <Row className="mt-3">
+          <Form.Group controlId="amount" as={Col}>
+            <InputGroup>
+              <InputGroup.Text>$</InputGroup.Text>
+              <Form.Control
+                type="number"
+                name="amount"
+                placeholder="Amount"
+                value={amount}
+                onChange={setValue}
+                autoComplete="off"
+              />
+            </InputGroup>
+          </Form.Group>
+          <Form.Group controlId="category" as={Col}>
+            <Form.Select name="category" value={category} onChange={setValue}>
+              <option value="" disabled selected>
```

Comment:
as `select` is controlled component the `selected` property is redundant here. The value of the component is controlled by `useState` hook and the default value is already set to `''`. Removing selected here won't affect functionality but  will also remove error in console browser.